### PR TITLE
Refactor `WalletPanel` and `SignatureProcessModal`. Fix `NavBar`, `NavBarMobile`, `TickerSelectorBox`, `CryptoCard` and `CryptoNews`, Add `User`.

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] 004	User
- [x] 104	UserOverview
- [x] 114	TradeVisitorTab
- [x] 043	OrderSection
- [x] 013	WalletPanel
- [x] 016	SignatureProcessModal
- [x]  ConnectingModal (LoadingModal)
- [x] 018	HelloModal
- [x] 017	QrcodeModal
- [x] 001	NavBar
- [x] 117	NavBarMobile
- [x] GlobalContext
- [x] 032	CryptoCategory
- [x] 033	CryptoCard
- [x] 037	TickerSelectorBox
- [x] 041	CryptoNewsSection
- [x] 042	CryptoNewsItem

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 1
src/components/user/user.tsx
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/wallet_panel/wallet_panel.tsx [hunk 1]
```diff
@@ -1491,83 +1367,155 @@ export default function WalletPanel({className, getUserLoginState}: IWalletPanel
...
-      if (chainId !== 1) {
-        // console.log('Please switch to ETH mainnet');
-        setShowToast(true);
+        if (connectWalletResult) {
+          globalCtx.visibleSignatureProcessModalHandler();
+        } else {
+          globalCtx.visibleWalletPanelHandler();
+        }
+      } catch (error) {
+        // console.error(error);
```
Comment (arealclimber) [inline src/components/wallet_panel/wallet_panel.tsx:1393]: Show up WalletPanel

File: src/components/signature_process_modal/signature_process_modal.tsx [hunk 2]
```diff
@@ -82,16 +140,38 @@ const SignatureProcessModal = ({
...
+    if (!userCtx.isConnected) {
+      // It's a cycle
+      const connectWalletResult = await userCtx.connect();
+    } else {
+      setConnectingProcess(ConnectingProcess.CONNECTING);
+      const signResult = await userCtx.signServiceTerm();
+      if (signResult) {
+        setTimeout(() => {
+          globalCtx.visibleSignatureProcessModalHandler();
+        }, 1000);
```
Comment (arealclimber) [inline src/components/signature_process_modal/signature_process_modal.tsx:153]: 在 common.ts 寫 Promise，專門等待 n 秒
```
(dosomething)
await wait(1000)
(dosomething)
```

File: src/components/signature_process_modal/signature_process_modal.tsx [hunk 3]
```diff
@@ -82,16 +140,38 @@ const SignatureProcessModal = ({
...
-  ) : (
-    <TideButton
-      onClick={requestSendingHandler}
-      className="rounded bg-tidebitTheme px-5 py-2 text-base transition-all hover:opacity-90"
-    >
-      Send Requests
-    </TideButton>
-  );
+  const requestSendingHandler = async () => {
+    if (!userCtx.isConnected) {
```
Comment (arealclimber) [inline src/components/signature_process_modal/signature_process_modal.tsx:144]: - 之後壓力測試再改 `setConnectingProcess(ConnectingProcess.CONNECTING);`

File: src/components/signature_process_modal/signature_process_modal.tsx [hunk 4]
```diff
@@ -82,16 +140,38 @@ const SignatureProcessModal = ({
     </div>
   );
```
Comment (arealclimber) [inline src/components/signature_process_modal/signature_process_modal.tsx:142]: - 寫在 common.ts 
```
let ifInProcess = false
function () {
ifInProcess = true
....
ifInProcess = false
}
```

Comment (arealclimber) [reply src/components/signature_process_modal/signature_process_modal.tsx:142]: 使用 lockerFunction
```tsx
const handler = () => {
  const [lock, unlock] = functionLocker('handler')
  if(!lock()) return;
  ...doSomeThing
  unlock();
}
```

Comment (arealclimber) [reply src/components/signature_process_modal/signature_process_modal.tsx:142]: 寫 lockerFunction
```tsx
const room = {};
const functionLocker = (roomId) => {
  const lock = () => {
    if(room[roomId]) {
      // 有人
      return false;
    } else {
      room[roomId] = true;
      return true;
    }
  }

  const unlock = () => {
    if(room[roomId]) {
      room[roomId] = false;
      return true;
    } else {
      throw new Error('流程有問題');
      return false;
    }
  }

  return [lock, unlock];
}
```

File: src/contexts/global_context.tsx [hunk 5]
```diff
@@ -461,5 +566,18 @@ export const useGlobal = () => {
...
+  const g: any =
+    typeof globalThis === 'object'
+      ? globalThis
+      : typeof window === 'object'
+      ? window
+      : typeof global === 'object'
+      ? global
+      : null; // Causes an error on the next line
+
+  g.gc = context;
```
Comment (arealclimber) [inline src/contexts/global_context.tsx:580]: `g.globalCtx = context`

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**